### PR TITLE
erlang-language-platform: init at 2024-07-16

### PR DESCRIPTION
**Pull request by offsetcyan**
## Description of changes

Adds the Erlang development tool [Erlang Language Platform](https://github.com/WhatsApp/erlang-language-platform) to nixpkgs, which involves patching the latest release binary.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/er/erlang-language-platform/package.nix
Comment (xfnw) [inline pkgs/by-name/er/erlang-language-platform/package.nix:43]: using `with lib;` by itself is [discouraged](https://github.com/NixOS/nixpkgs/issues/208242)
```suggestion
  meta = {
    description = "An IDE-first library for the semantic analysis of Erlang code, including LSP server, linting and refactoring tools.";
    homepage = "https://github.com/WhatsApp/erlang-language-platform/";
    license = with lib.licenses; [
```

Comment (offsetcyan) [reply pkgs/by-name/er/erlang-language-platform/package.nix:43]: Fixed, thanks for a link to the discussion.

File: pkgs/by-name/er/erlang-language-platform/package.nix
Comment (chiroptical) [inline pkgs/by-name/er/erlang-language-platform/package.nix:24]: Does this need to exist under the OTP 26 erlang version? I can't recall off-hand if that matters.

**Review by chiroptical — APPROVED** [review]
This looks right to me, but you may need to tag another reviewer. I added one question because I can't recall how ELP behaves if you are on OTP 25 or 27.

**Comment by chiroptical** [comment]
@happysalada any concerns on this one?

(note: I keep tagging you because you seem to be the most active erlang committer in my experience, let me know if it annoying please)

**Comment by happysalada** [comment]
@chiroptical all good, please tag @NixOS/beam so that everyone can get a chance to see it

**Pull request merged**
